packages/vue-server-renderer/types/index.d.ts

```diff
@@ -0,0 +1,44 @@
+import Vue = require('vue');
+import { Readable } from 'stream';
+
+export declare function createRenderer(options?: RendererOptions): Renderer;
+
+export declare function createBundleRenderer(bundle: any, options?: BundleRendererOptions): BundleRenderer;
+
+type RenderCallback = (err: Error, html: string) => void;
+
+interface Renderer {
+  renderToString(vm: Vue, callback: RenderCallback): void;
+  renderToString(vm: Vue, context: object, callback: RenderCallback): void;
+
+  renderToStream(vm: Vue, context?: object): Readable;
+}
+
+interface BundleRenderer {
+  renderToString(callback: RenderCallback): void;
+  renderToString(context: object, callback: RenderCallback): void;
+
+  renderToStream(context?: object): Readable;
+}
+
+interface RendererOptions {
+  template?: string;
+  inject?: boolean;
+  shouldPreload?: (file: string, type: string) => boolean;
```

Comment:
We can even refine `type` to a precise list of file type listed [here](https://fetch.spec.whatwg.org/#concept-request-destination).

Might be an overkill.